### PR TITLE
ref(ui): Add performance event view hook

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
This adds a hook to avoid having to constantly drill event view, especially for our generic discover query specializations. Essentially if any specialization now uses the 'DiscoverQueryPropsWithContext' type you no longer have to provide eventView and orgSlug, which should reduce a lot of boilerplate throughout performance. It's not on by default since there is likely some undefined checks and tests that might need to be rewritten (to use a provider) if you choose to use this.

Other:
- Giving a context wrapper to always check if a context is defined, we'll see if it's helpful, and if so we can raise this up to be used everywhere in the UI or talk about it during TSC.
- This will currently allow `usePerformanceEventView` to be used under the main performance content page and vital details, we can add it to the rest in subsequent PR's.

### CODE REVIEW

File: static/app/views/performance/contexts/performanceEventViewContext.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+// Provides a readonly event view. Also omits anything that isn't currently read-only, although in the future we should switch the code in EventView instead.
+// If you need mutability, use the mutable version.
+const usePerformanceEventView: () => ImmutableEventView = () =>
+  _usePerformanceEventView().eventView;
+export {PerformanceEventViewProvider, usePerformanceEventView};
+
+export function useMutablePerformanceEventView() {
+  return usePerformanceEventView().clone();
+}
```
Comment (Zylphrex) [inline static/app/views/performance/contexts/performanceEventViewContext.tsx:22]: Nit: this just looks inconsistent with the mix of arrow functions, normal functions, and the different syntaxes for exports.

File: static/app/views/performance/contexts/utils.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
```
Comment (Zylphrex) [inline static/app/views/performance/contexts/utils.tsx:1]: Should this file exist within `static/app/utils/...` instead?

**Review by Zylphrex — APPROVED** [review]

File: static/app/views/performance/contexts/utils.tsx (hunk 2 above)
Comment (k-fish) [reply static/app/views/performance/contexts/utils.tsx:1]: Definitely, good call.

File: static/app/views/performance/contexts/performanceEventViewContext.tsx (hunk 1 above)
Comment (k-fish) [reply static/app/views/performance/contexts/performanceEventViewContext.tsx:22]: all-the-things.jpg

File: static/app/utils/discover/genericDiscoverQuery.tsx [hunk 3]
```diff
@@ -228,6 +244,19 @@ class GenericDiscoverQuery<T, P> extends React.Component<Props<T, P>, State<T>>
   }
 }
 
+// Shim to allow us to use generic discover query or any specialization with or without passing org slug or eventview, which are now contexts.
+// This will help keep tests working and we can remove extra uses of context-provided props and update tests as we go.
```
Comment (calvinlough) [inline static/app/utils/discover/genericDiscoverQuery.tsx:248]: Good call

Comment (k-fish) [reply static/app/utils/discover/genericDiscoverQuery.tsx:248]: Thanks.

Comment (dashed) [reply static/app/utils/discover/genericDiscoverQuery.tsx:248]: 🤔

**Pull request merged**